src/biome/text/tokenizer.py

```diff
@@ -20,61 +21,47 @@ class Tokenizer(FromParams):
-    max_sequence_length: `int`
-        Maximum length in characters for input texts truncated with `[:max_sequence_length]` after `TextCleaning`.
-    max_nr_of_sentences: `int`
-        Maximum number of sentences to keep when using `segment_sentences` truncated with `[:max_sequence_length]`.
-    text_cleaning: `Optional[TextCleaning]`
-        A `TextCleaning` configuration with pre-processing rules for cleaning up and transforming raw input text.
-    segment_sentences:  `Union[bool, SentenceSplitter]`
-        Whether to segment input texts in to sentences using the default `SentenceSplitter` or a given splitter.
-    start_tokens: `Optional[List[str]]`
-        A list of token strings to the sequence before tokenized input text.
-    end_tokens: `Optional[List[str]]`
-        A list of token strings to the sequence after tokenized input text.
+    config
+        A `TokenizerConfiguration` object
     """
-
-    # TODO(dcfidalgo): Check if it is possible to pass in a `TokenizerConfiguration` to avoid duplication of arguments
-
     def __init__(
         self,
-        lang: str = "en",
-        max_sequence_length: int = None,
-        max_nr_of_sentences: int = None,
-        text_cleaning: Optional[TextCleaning] = None,
-        segment_sentences: Union[bool, SentenceSplitter] = False,
-        start_tokens: Optional[List[str]] = None,
-        end_tokens: Optional[List[str]] = None,
+        config: TokenizerConfiguration,
     ):
-        self._fetch_spacy_model(lang)
+        self.lang = config.lang
+        self._fetch_spacy_model()
+
+        if config.segment_sentences is True:
+            self.segment_sentences = SpacySentenceSplitter(language=self.lang, rule_based=True)
+        elif config.segment_sentences is False:
+            self.segment_sentences = None
+        else:
+            self.segment_sentences = SentenceSplitter.from_params(Params(config.segment_sentences))
+
+        self.max_nr_of_sentences = config.max_nr_of_sentences
 
-        if segment_sentences is True:
-            segment_sentences = SpacySentenceSplitter(language=lang, rule_based=True)
+        self.max_sequence_length = config.max_sequence_length
 
-        self.lang = lang
-        self.segment_sentences = segment_sentences
-        self.max_nr_of_sentences = max_nr_of_sentences
-        self.max_sequence_length = max_sequence_length
-        self.text_cleaning = text_cleaning or DefaultTextCleaning()
+        if config.text_cleaning is None:
+            self.text_cleaning = DefaultTextCleaning()
+        else:
+            self.text_cleaning = TextCleaning.from_params(Params(config.text_cleaning))
 
         self._base_tokenizer = SpacyTokenizer(
-            language=self.lang, start_tokens=start_tokens, end_tokens=end_tokens,
+            language=self.lang, start_tokens=config.start_tokens, end_tokens=config.end_tokens,
         )
 
-    def _fetch_spacy_model(self, lang):
+    def _fetch_spacy_model(self):
```

Comment:
I prefer to keep language as required argument and decorate method as @staticmethod  if you don't mind

The following code could be read as good since we really don't known anything about the `self.lang` dependency in `_fetch_spacy_model`
```python
self._fetch_spacy_model()
self.lang = lang
```

Reply:
done